Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Debug.azsli

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/Features/SrgSemantics.azsli>
+
+#include <scenesrg.srgi>
+#include <viewsrg.srgi>
+
+#include <Atom/RPI/BitOperations.azsli>
+
+#ifndef ENABLE_SHADER_DEBUGGING
+#define ENABLE_SHADER_DEBUGGING     1
+#endif
+
+#if ENABLE_SHADER_DEBUGGING
+    option bool o_shader_debugging_enabled = true;
+#else
+    #define o_shader_debugging_enabled false
+#endif
+
+// Options controlling various aspects of the rendering
+enum class RenderDebugOptions
+{
+    DebugEnabled,
+
+    OverrideBaseColor,
+    OverrideRoughness,
+    OverrideMetallic,
+
+    UseDebugLight,
+
+    EnableNormalMaps,
+    EnableDetailNormalMaps,
+
+    EnableDiffuseLighting,
+    EnableSpecularLighting,
+    EnableDirectLighting,
+    EnableIndirectLighting,
+};
+
+// Specifies what debug info to render to the view
+enum class RenderDebugViewMode
+{
+    None,
+
+    Normal,
+    Tangent,
+    Bitangent,
+    
+    BaseColor,
+    Albedo,
+    Roughness,
+    Metallic,
+};
+
+bool IsDebuggingEnabled()
+{
+#if ENABLE_SHADER_DEBUGGING
+    return o_shader_debugging_enabled && SceneSrg::m_debuggingEnabled;
+#else
+    return false;
+#endif
+}
+
+// Placeholder version of the above function until o_shader_debugging_enabled can be hooked up to all shaders
+bool IsDebuggingEnabled_PLACEHOLDER()
+{
+#if ENABLE_SHADER_DEBUGGING
+    return SceneSrg::m_debuggingEnabled;
+    //return o_shader_debugging_enabled && SceneSrg::m_debuggingEnabled;
```

Comment:
Commented code, intentional?  Minor issue since this function is probably going away at some point.

Reply:
The commented line is what it should be, keeping it there so I know what to revert it to when I test and implement the shader option part.